app/controllers/posts_controller.rb

```diff
@@ -2,23 +2,41 @@
 
 class PostsController < ApplicationController
   def index
-    @posts = Post.all
+    top_post_id = Post.joins(:votes).group(:post_id).count.sort.to_h
+    @top_post = if top_post_id.nil?
+                  Post.find(top_post_id.first[0])
+                else
+                  Post.last
+                end
+    @posts = Post.where.not(id: top_post_id)
+
   end
 
   before_action :authenticate_author!, only: %i[new create]
   def new
     @post = Post.new
+    @categories = Category.all.map{ |cat| [cat.name, cat.id]}
   end
 
   def show
     @post = Post.find(params[:id])
     @author = Author.find(@post.author_id)
+    @votes = Vote.where(post_id: params[:id])
+    @score = 0
+    @votes.each do |vote|
```

Comment:
Cały kod który liczy score to logika biznesowa, która nie powinna być w modelu. Proponuję dodać publiczną metodę do modelu `Post`, która to liczy:
```ruby
class Post 

[...]

  def score
    result = 0
    votes.each do |vote|
      if vote.vote_type == 0
        result += 1
      else
        result -= 1
      end
    end
    result
  end

[...]
end
```

wtedy też będzie można przetestować taką metodę jednostkowo